Bizyhood/Core.php

```diff
@@ -1816,6 +1826,67 @@ public function search_shortcode($attrs) {
     }
     
     
+    
+    
+    
+    public function businesses_group_request($groupd_id)
+    {
+      
+      $client = Bizyhood_oAuth::oAuthClient();
+           
+      if (is_wp_error($client)) {
+        $error = new WP_Error( 'bizyhood_error', $client->get_error_message() );
+        return array('error' => $error);
+      }      
+      
+      
+      $api_url = Bizyhood_Utility::getApiUrl();
+      
+      try {
+        $response = $client->fetch($api_url.'/business/group/'. $groupd_id .'/');
+      } catch (Exception $e) {
+        $error = new WP_Error( 'bizyhood_error', __( 'Service is currently unavailable! Request timed out.', 'bizyhood' ) );
+        return array('error' => $error);
+      }  
+      
+      // avoid throwing an error
+      if (!is_array($response) || (is_array($response) && isset($response['code']) && $response['code'] != 200)) { return false; }
+      
+      return $response;
+      
+    }
+    public function businesses_group($attrs)
+    {
+      $attributes = shortcode_atts( array(
+        'group' => false, // group id 
+      ), $attrs );
+    
+      
+      $authetication = Bizyhood_oAuth::set_oauth_temp_data();
+      if (is_wp_error($authetication)) {
+        return Bizyhood_View::load( 'listings/error', array( 'error' => $authetication->get_error_message()), true );
+      }
+      
+      if (Bizyhood_oAuth::checkoAuthData() == false || get_transient('bizyhood_oauth_data') === false) {
+        return Bizyhood_View::load( 'listings/error', array( 'error' => 'oAuth Failed'), true );
+      }
+      if ($attrs['group'] === false) {
+        return Bizyhood_View::load( 'listings/error', array( 'error' => 'The Group ID is required'), true );
```

Comment:
This line is repeated a lot of times. I think it makes sense to check all the conditions and set the correct error message first, then just call this line once, otherwise this gets needlessly verbose.